File: backend/MiniBilge.Application/DTOs/Profile/ChildProfileDto.cs
namespace MiniBilge.Application.DTOs.Profile;

public class ChildProfileDto
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public DateTime DateOfBirth { get; set; }
    public int Age { get; set; }
    public string GradeLevel { get; set; } = string.Empty;
    public string ProfileType { get; set; } = "Child";
    public string? EnglishLevel { get; set; }
    public string? AvatarImageUrl { get; set; }   // Hesaplanan: preference'a göre fotoğraf veya karakter
    public string? PhotoUrl { get; set; }           // Ham fotoğraf URL'si (kayıtlıysa)
    public bool UseAvatarAsProfile { get; set; }    // true = avatar göster, false = fotoğraf göster
    public int TotalCoins { get; set; }
    public int TotalStars { get; set; }
    /// <summary>0 = Offline (cihaz TTS), 1 = Online (Google TTS)</summary>
    public int PodcastListeningMode { get; set; } = 0;
    /// <summary>Arkadaş davetlerinde paylaşılan 6 karakterli benzersiz kod (MB-XXXXXX).</summary>
    public string FriendCode { get; set; } = string.Empty;
    public bool IsTeacher { get; set; }
}

File: mobile/minibilge_mobile/lib/features/child_profile/models/child_profile_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'grade_level.dart';
import 'english_level.dart';

part 'child_profile_dto.freezed.dart';
part 'child_profile_dto.g.dart';

@freezed
class ChildProfileDto with _$ChildProfileDto {
  const factory ChildProfileDto({
    @JsonKey(name: 'Id') required String id,
    @JsonKey(name: 'Name') required String name,
    @JsonKey(name: 'DateOfBirth') required DateTime dateOfBirth,
    @JsonKey(name: 'Age') required int age,
    @JsonKey(name: 'GradeLevel') required String gradeLevel,
    @JsonKey(name: 'EnglishLevel') String? englishLevel,
    @JsonKey(name: 'AvatarImageUrl') String? avatarImageUrl,
    @JsonKey(name: 'TotalCoins') @Default(0) int totalCoins,
    @JsonKey(name: 'TotalStars') @Default(0) int totalStars,
    // 0 = Offline (cihaz TTS), 1 = Online (Google TTS)
    @JsonKey(name: 'PodcastListeningMode') @Default(0) int podcastListeningMode,
    @JsonKey(name: 'FriendCode') String? friendCode,
    @JsonKey(name: 'IsTeacher') @Default(false) bool isTeacher,
  }) = _ChildProfileDto;

  factory ChildProfileDto.fromJson(Map<String, dynamic> json) =>
      _$ChildProfileDtoFromJson(json);
}

extension ChildProfileDtoX on ChildProfileDto {
  GradeLevel? get gradeLevelEnum => GradeLevel.fromString(gradeLevel);
  EnglishLevel? get englishLevelEnum => EnglishLevel.fromString(englishLevel);
}
